File: aoc2021/21-dirac-dice/dirac_dice.c
#include <stdio.h>

int die = 1;
int rolls = 0;

int main(void) {
    int players[2] = {4, 1};
    int scores[2] = {0, 0};

    int player = 0;
    while (scores[0] < 1000 && scores[1] < 1000) {
        int move = die + die+1 + die+2;
        die += 3;
        rolls += 3;
        players[player] = ((players[player] + move - 1) % 10) + 1;
        scores[player] += players[player];
        printf("player %d rolls %d and moves to space %d for total score of %d\n",
            player+1, move, players[player], scores[player]);
        player = !player;
    }

    printf("Part one result: %d\n", scores[player] * rolls);

    return 0;
}

File: aoc2021/21-dirac-dice/dirac_dice.py

die = 1
rolls = 0


if __name__ == "__main__":
    players = [4, 1]
    scores = [0, 0]

    player = 0
    while scores[0] < 1000 and scores[1] < 1000:
        move = die + die+1 + die+2
        die += 3
        rolls += 3
        players[player] = ((players[player] + move - 1) % 10) + 1
        scores[player] += players[player]
        # print(f"player {player+1} rolls {move} and moves to space {players[player]} for total score of {scores[player]}")
        player = not player

    print(scores[player] * rolls)








